drf_spectacular/openapi.py

```diff
@@ -165,6 +165,11 @@ def dict_helper(parameters):
         # override/add @extend_schema parameters
         for key, parameter in override_parameters.items():
             parameters[key] = parameter
+
+        format_param = parameters.get(('format', 'path'))
```

Comment:
this has multiple problems

1. if there is no enum there, you cannot always enforce the format enum through the renderers. manually handling the format param in the view would be impossible like that, because the param would have fixed choices
2. `format_param['enum']` violates the json specification. -> format_param['schema']['enum']
3. the `f` is missing so the literal string gets added

i think this has to go (map_formats as well)

Reply:
> You cannot always enforce the format enum through the renderers.

Why not?  This is DRF core standard behaviour, and I explained this in the issue.

This list of formats is derived from the configured renderers.  Full stop.  Trying to put a format in a URL results in 404 if that format doesnt exist, i.e. registered by some renderer.  The view never sees an invalid `format` value that isnt associated with a renderer.

I guess you are seeing the regex noted at https://github.com/tfranzel/drf-spectacular/pull/123#discussion_r453598774 allows any alphanum string through during param mapping, however that is just because DRF doesnt want to do the work of getting the real list *there*, because invalid values are rejected later.

Anyways, you pushing back here meant I had to double check, and also now reject unavailable formats even if they are in theory going to get through the URLConf phase, because they will not be valid endpoints.  I guess the DRF functionality exists this way so that an installable app can support multiple formats, but the project can supports a smaller subset.

I havent done any special handling for the crazy scenario that an app uses only formats which are not available, which would make the entire format arg unusable, and because the format arg appears in a duplicated endpoint, the entire additional endpoint becomes invalid.  Removing `{format}` endpoints is legitimate, but gets beyond this PRs scope.

Reply:
yes you are right. i was not aware of that behaviour and in fact a bit surprised. the doc is kind of foggy on that subject. had to check with a test myself. the generic regex threw me off.

> doesnt want to do the work of getting the real list there, because invalid values are rejected later.

thats seems to be the case

fyi: The `URL_FORMAT_OVERRIDE` seems to only apply to the format query param (and not the path param). we can think about handling that later.

i think we are done here.thanks. i'll give it another quick look and then merge.